public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -81,6 +81,20 @@ export class PrometheusDatasource {
     return this.backendSrv.datasourceRequest(options);
   }
 
+  // Use this for tab completion features, wont publish response to other components
+  helperRequest(url) {
+    const options: any = {
+      url: this.url + url,
+      silent: true,
+    };
+
+    if (this.basicAuth || this.withCredentials) {
```

Comment:
It's a bit unfortunate to multiply this logic and possibly easy to miss if making a change to the _request function. Would it be possible to let this function reuse the _request function? Maybe by introducing a new argument at the end of _request function.